lib/CL/devices/pthread/pthread_scheduler.c

```diff
@@ -205,10 +207,12 @@ get_wg_index_range (kernel_run_command *k, unsigned *start_index,
    * num_threads, otherwise fallback to smaller workgroups.
    */
   if (k->remaining_wgs <= (scaled_max_wgs * num_threads))
-    max_wgs = min (scaled_min_wgs, (1 + k->remaining_wgs / num_threads));
+    limit = scaled_min_wgs;
   else
-    max_wgs = min (scaled_max_wgs, (1 + k->remaining_wgs / num_threads));
+    limit = scaled_max_wgs;
 
+  const unsigned wgs_per_thread = ceil ((float) k->remaining_wgs / (float) num_threads);
```

Comment:
I think a manual `ceil()` using integer-arithmetic is a cleaner solution here:

https://stackoverflow.com/questions/2745074/fast-ceiling-of-an-integer-division-in-c-c